client/utils/dateInterval.js

```diff
@@ -0,0 +1,30 @@
+export const calculateDateInterval = (startDate, endDate) => {
+    let intervalUnit = '';
+    let intervalNumber = 0;
+    const dateDifferenceMilliseconds = endDate - startDate;
+    const timeIntervalInHours = Math.floor(
+        dateDifferenceMilliseconds / (1000 * 60 * 60)
+    );
+    const timeIntervalInDays = Math.floor(timeIntervalInHours / 24);
+
+    if (timeIntervalInHours < 24) {
+        intervalNumber = timeIntervalInHours;
+        intervalUnit = 'hour';
+    } else {
+        intervalUnit = 'day';
+        intervalNumber = timeIntervalInDays;
+    }
+
+    if (intervalNumber > 1) {
+        intervalUnit = `${intervalUnit}s`;
+    }
+
+    if (timeIntervalInDays >= 30) {
+        const dateString = `${startDate.toLocaleString('default', {
+            month: 'short'
+        })} ${startDate.getDate()}, ${startDate.getFullYear()}`;
+        return dateString;
+    }
+
+    return `${intervalNumber} ${intervalUnit} ago`;
+};
```

Comment:
Nicely done!

Reply:
agreed with @alflennik that this is really slick! However, Matt prefers that we use absolute dates instead of relative times, always (I mentioned that GitHub uses this approach, but he felt consistency and no relative times was clearer). Can we change the output to always be the formatted date string?

Also, can we add the time in order to disambiguate between two commits on the same day?

What do you think of something like this?
 
```js
`${startDate.toLocaleString('default', { month: 'short' })} ${startDate.getDate()}, ${startDate.getFullYear()} ${startDate.toLocaleTimeString();}`
```